app/wizards/placements/add_partnership_wizard.rb

```diff
@@ -0,0 +1,46 @@
+module Placements
+  class AddPartnershipWizard < BaseWizard
+    attr_reader :organisation, :partner_organisation_model
+
+    def initialize(organisation:, params:, session:, current_step: nil)
+      @organisation = organisation
+      @partner_organisation_model = partner_org_model
+      super(session:, params:, current_step:)
+    end
+
+    def define_steps
+      add_step(PartnershipStep)
+      add_step(PartnershipOptionsStep) if steps[:partnership].partner_organisation.blank?
+      add_step(CheckYourAnswersStep)
+    end
+
+    def partner_organisation
+      @partner_organisation ||= (steps[:partnership_options]&.partner_organisation ||
+          steps[:partnership].partner_organisation).decorate
+    end
+
+    def create_partnership
+      raise "Invalid wizard state" unless valid?
+
+      if organisation.respond_to?(:partner_schools)
+        organisation.partner_schools << partner_organisation
+      else
+        organisation.partner_providers << partner_organisation
+      end
+      organisation.save!
+    end
+
+    def partner_organisation_type
+      @partner_organisation_type ||= partner_organisation_model.to_s.downcase
+    end
+
+    private
+
+    def partner_org_model
+      @partner_org_model ||= {
+        School => ::Provider,
+        Provider => ::School,
+      }.fetch(organisation.class)
+    end
```

Comment:
I don't think we need both `partner_organisation_model` and `partner_org_model` defined as separate things. I think they can be one and the same.

If you rename this method to `partner_organisation_model`:

```suggestion
    def partner_organisation_model
      @partner_organisation_model ||= {
        School => ::Provider,
        Provider => ::School,
      }.fetch(organisation.class)
    end
```

Then you won't need the `attr_reader` or the line in the initialize method:

```diff
- attr_reader :organisation, :partner_organisation_model
+ attr_reader :organisation

def initialize(organisation:, params:, session:, current_step: nil)
  @organisation = organisation
- @partner_organisation_model = partner_org_model
  super(session:, params:, current_step:)
end
```